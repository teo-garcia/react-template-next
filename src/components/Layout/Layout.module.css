.layout {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
}

.main {
  grid-column: 1/ -1;
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
}
