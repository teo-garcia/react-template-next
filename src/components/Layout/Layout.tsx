import styles from './Layout.module.css';
import { LayoutProps } from '../../types/client';

function Layout(props: LayoutProps) {
  const { children } = props;
  return (
    <div className={styles.layout}>
      <header />
      <main className={styles.main}>{children}</main>
      <footer />
    </div>
  );
}

export { Layout };
