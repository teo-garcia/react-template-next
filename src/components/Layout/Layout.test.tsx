import { render, screen } from '@testing-library/react';
import { Layout } from './Layout';

describe('<Layout /> tests', function () {
  test('Should match with the component snapshot', function () {
    const { asFragment } = render(<Layout>Hello World</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should render the provided children prop', function () {
    render(<Layout>Hello World</Layout>);
    expect(screen.getByText(/Hello World/i)).toBeInTheDocument();
  });
});
