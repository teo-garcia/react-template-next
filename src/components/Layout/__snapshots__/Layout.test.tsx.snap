// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> tests Should match with the component snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header />
    <main>
      Hello World
    </main>
    <footer />
  </div>
</DocumentFragment>
`;
